cmd/exo/cmd/firewall_remove.go

```diff
@@ -115,33 +134,25 @@ func removeAllRules(sgName string) error {
 	return nil
 }
 
-func removeRule(name, ruleID string) error {
-	sg, err := getSecurityGroupByNameOrID(name)
-	if err != nil {
-		return err
-	}
-
+func removeRule(sg *egoscale.SecurityGroup, ruleID string, tasks *[]task) error {
 	id, err := egoscale.ParseUUID(ruleID)
 	if err != nil {
 		return err
 	}
 
 	in, eg := sg.RuleByID(*id)
 
+	var msg string
 	if in != nil {
-		err = cs.BooleanRequestWithContext(gContext, &egoscale.RevokeSecurityGroupIngress{ID: in.RuleID})
+		msg = fmt.Sprintf("Remove %q", in.RuleID)
+		*tasks = append(*tasks, task{egoscale.RevokeSecurityGroupIngress{ID: in.RuleID}, msg})
 	} else if eg != nil {
-		err = cs.BooleanRequestWithContext(gContext, &egoscale.RevokeSecurityGroupEgress{ID: eg.RuleID})
+		msg = fmt.Sprintf("Remove %q", eg.RuleID)
+		*tasks = append(*tasks, task{egoscale.RevokeSecurityGroupEgress{ID: eg.RuleID}, msg})
 	} else {
-		err = fmt.Errorf("rule with id %q is not ingress or egress rule", ruleID)
+		return fmt.Errorf("rule with id %q is not ingress or egress rule", ruleID)
```

Comment:
it more that the rule doesn't exist.

https://github.com/exoscale/egoscale/blob/master/security_groups.go#L71